ui/apps/platform/src/services/GroupsService.ts

```diff
@@ -19,16 +20,16 @@ export function fetchGroups() {
  *
  * @returns {Promise<Object, Error>}
  */
-export function updateOrAddGroup({ oldGroups, newGroups }) {
-    return axios.post(updateUrl, { previous_groups: oldGroups, required_groups: newGroups });
+export function updateOrAddGroup(request: GroupBatchUpdateRequest) {
+    return axios.post(updateUrl, request);
 }
 
 /**
  * Fetches the default rule.
  *
  * @returns {Promise<Object, Error>} fulfilled default group
  */
-export function getDefaultGroup({ authProviderId, roleName }) {
+export function getDefaultGroup({ authProviderId, roleName }: GetDefaultGroupRequest) {
```

Comment:
You can add `: Promise<Group | undefined>` as return type.

Reply:
Unfortunately, not, the third type that is suggested to me is `{ response: any; }`

Reply:
What does it think about `get<{ groups: Group[] }>(…)`

Reply:
Then
```
Type 'Promise<Group | { response: Group; } | undefined>' is not assignable to type 'Promise<Group | undefined>'
```

If I put `{ response: {groups: Group[] }}` then
```
TS2322: Type 'Promise<Group | { response: any; } | undefined>' is not assignable to type 'Promise<Group | undefined>'
```